tests/suites/test_suite_pk.function

```diff
@@ -216,7 +216,7 @@ void pk_psa_utils(int key_is_rsa)
     size_t len;
     mbedtls_pk_debug_item dbg;
 
-    PSA_ASSERT(psa_crypto_init());
+    USE_PSA_INIT();
```

Comment:
Pre-existing here, new in other places if `psa_crypto_init()` fails, this will jump to the exit label, which will call `pk_free()` - if the contexts have not been initialized yet, then this will result in using uninitialized pointers, which even on an error path in tests is not what we want.

The simple fix is to make sure that all structures that are `mbedtls_xxx_free()`d at `exit` have been `mbedtls_xxx_init()`ialized before the first thing that may jump to exit.

You'll probably point out that the documentation currently says we must call `psa_crypto_init()` before calling any function from the PK, X.509 or TLS module... and you'd be right! I think we should amend the documentation to append "except for the various mbedtls_xxx_init() functions which can be called at any time".